pymongo/typings.py

```diff
@@ -36,6 +35,5 @@
 _CollationIn = Union[Mapping[str, Any], "Collation"]
 _DocumentIn = Union[MutableMapping[str, Any], "RawBSONDocument"]
 _Pipeline = Sequence[Mapping[str, Any]]
-_DocumentType = TypeVar(
-    "_DocumentType", Mapping[str, Any], MutableMapping[str, Any], Dict[str, Any]
-)
+_DocumentOut = _DocumentIn
+_DocumentType = TypeVar("_DocumentType", bound=Mapping[str, Any])
```

Comment:
I cleaned this up to be `_DocumentType = TypeVar("_DocumentType", MutableMapping[str, Any], "RawBSONDocument")` now that we're not confined by the default value limitation.

Reply:
Wait I thought `TypeVar("_DocumentType", bound=Mapping[str, Any])` was the better option? In my testing it supports TypedDict and everything else. The only quirk is that user will need to add type annotations for their MongoClients:
```
generic.py:7: error: Need type annotation for "client"  [var-annotated]
        client = MongoClient()
                 ^
```
Thats not so bad since list/dict/set/etc all have the same requirement: https://mypy.readthedocs.io/en/stable/type_inference_and_annotations.html#explicit-types-for-collections

And in the future when mypy gets their act together we can add `dict` as the default. 

One other quirk is that it leads to this internal pymongo mypy error:
```
mypy pymongo                                     
pymongo/mongo_client.py: note: In member "__init__" of class "MongoClient":
pymongo/mongo_client.py:127: error: Incompatible default for argument "document_class" (default has type "Type[Dict[Any, Any]]", argument has type
"Type[_DocumentType]")  [assignment]
            document_class: Type[_DocumentType] = dict,
                                                  ^
```

We could fix this by changing it to `document_class: Optional[Type[_DocumentType]] = None` instead of dict.

Reply:
Updated

Reply:
Ack.